plugins/catalog/src/components/CatalogTable/CatalogTable.tsx

```diff
@@ -97,8 +98,10 @@ export const CatalogTable = (props: CatalogTableProps) => {
     emptyContent,
   } = props;
   const { isStarredEntity, toggleStarredEntity } = useStarredEntities();
-  const entityListContext = useEntityList();
 
+  const { allKinds } = useAllKinds();
+
+  const entityListContext = useEntityList();
```

Comment:
This does make me uneasy. Especially for users with huge catalogs, we now make this extra work/request mandatory and happen on every catalog table everywhere. And additionally, it's work that's already being done elsewhere (by the actual kind filter) so it's duplucated and wasteful.

You already are reading `filters.kind?.value`. Is the problem that that one doesn't have the right casing? Can we make it have the right case in the first place instead? Or have an additional special field `originalCaseValue` if we're forced to keep the old behavior for some reason - but I don't thing we are.

Reply:
I have reworked this significantly and it no longer requires any additional calls.